TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/RequestSenderOrchestrator.java

```diff
@@ -0,0 +1,239 @@
+package org.opensearch.migrations.replay;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.EventLoop;
+import lombok.extern.slf4j.Slf4j;
+import org.opensearch.migrations.replay.datahandlers.NettyPacketToHttpConsumer;
+import org.opensearch.migrations.replay.datatypes.ChannelAndScheduledRequests;
+import org.opensearch.migrations.replay.datatypes.UniqueRequestKey;
+import org.opensearch.migrations.replay.util.DiagnosticTrackableCompletableFuture;
+import org.opensearch.migrations.replay.util.StringTrackableCompletableFuture;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+@Slf4j
+public class RequestSenderOrchestrator {
+
+    private static class TimeAdjustData {
+        public final Instant firstTimestamp;
+        public final Instant systemTimeForFirstReplay;
+
+        public TimeAdjustData(Instant firstTimestamp) {
+            this.firstTimestamp = firstTimestamp;
+            this.systemTimeForFirstReplay = Instant.now();
+        }
+    }
+
+    // TODO - Reconsider how time shifting is done
+    AtomicReference<TimeAdjustData> firstTimestampRef = new AtomicReference<>();
+
+    public final ClientConnectionPool clientConnectionPool;
+
+    public RequestSenderOrchestrator(ClientConnectionPool clientConnectionPool) {
+        this.clientConnectionPool = clientConnectionPool;
+    }
+
+    Instant getTransformedTime(Instant ts) {
+        firstTimestampRef.compareAndSet(null, new TimeAdjustData(ts));
+        var rval = firstTimestampRef.get().systemTimeForFirstReplay
+                .plus(Duration.between(firstTimestampRef.get().firstTimestamp, ts).multipliedBy(2));
```

Comment:
What am I missing here with the `multipliedBy(2)`, say we had the following sequence at startup:
1. TS1 8:00 AM
2. TS2 9:00 AM

Wouldn't the replay sequence be:
1. TS1 transformed time: 8:00 AM + (0 * 2)
2. TS2 transformed time: 8:00 AM + (1 hour * 2)

Wouldn't TS2 get replayed an hour later than normal?